src/mongo/db/repl/initial_syncer_fcb.cpp

```diff
@@ -0,0 +1,2536 @@
+        return;
+    }
+} catch (const DBException&) {
+    // Report exception as an initial syncer failure.
+    stdx::unique_lock<Latch> lock(_mutex);
+    onCompletionGuard->setResultAndCancelRemainingWork_inlock(lock, exceptionToStatus());
+}
+
+void InitialSyncerFCB::_finalizeAndCompleteCallback(
+    const executor::TaskExecutor::CallbackArgs& callbackArgs,
+    // NOLINTNEXTLINE(*-unnecessary-value-param)
+    std::shared_ptr<OnCompletionGuard> onCompletionGuard) noexcept try {
+    stdx::lock_guard<Latch> lock(_mutex);
+    auto status = _checkForShutdownAndConvertStatus_inlock(
+        callbackArgs, "_finalizeAndCompleteCallback cancelled");
+    if (!status.isOK()) {
+        onCompletionGuard->setResultAndCancelRemainingWork_inlock(lock, status);
+        return;
+    }
+
+    {
+        auto opCtx = makeOpCtx();
+        // Attach JournalListener to the new instance of storage engine
+        auto* journalListener = _dataReplicatorExternalState->getReplicationJournalListener();
+        opCtx->getServiceContext()->getStorageEngine()->setJournalListener(journalListener);
+    }
+
+    // Successfully complete initial sync
+    onCompletionGuard->setResultAndCancelRemainingWork_inlock(lock, _lastApplied);
+} catch (const DBException&) {
+    // Report exception as an initial syncer failure.
+    stdx::unique_lock<Latch> lock(_mutex);
+    onCompletionGuard->setResultAndCancelRemainingWork_inlock(lock, exceptionToStatus());
+}
+
+// template
+// void InitialSyncerFCB::_(const executor::TaskExecutor::CallbackArgs& callbackArgs,
+//                         // NOLINTNEXTLINE(*-unnecessary-value-param)
+//                         std::shared_ptr<OnCompletionGuard> onCompletionGuard) noexcept try {
+//    stdx::lock_guard<Latch> lock(_mutex);
+//    auto status = _checkForShutdownAndConvertStatus_inlock(callbackArgs, "error message");
+//    if (!status.isOK()) {
+//        onCompletionGuard->setResultAndCancelRemainingWork_inlock(lock, status);
+//        return;
+//    }
+//
+//    // schedule next task
+//    status = _scheduleWorkAndSaveHandle_inlock(
+//        [this, onCompletionGuard](const executor::TaskExecutor::CallbackArgs& args) {
+//            _nextTaskCallback(args, onCompletionGuard);
+//        },
+//        &_currentHandle,
+//        "task name");
+//    if (!status.isOK()) {
+//        onCompletionGuard->setResultAndCancelRemainingWork_inlock(lock, status);
+//        return;
+//    }
+//} catch (const DBException&) {
+//    // Report exception as an initial syncer failure.
+//    stdx::unique_lock<Latch> lock(_mutex);
+//    onCompletionGuard->setResultAndCancelRemainingWork_inlock(lock, exceptionToStatus());
+//}
+
+// debugging template
+//            onCompletionGuard->setResultAndCancelRemainingWork_inlock(
+//                lock,
+//                {ErrorCodes::NotImplemented,
+//                 "All files cloned; cancel FCBIS for debugging reason"});
```

Comment:
I suggest removing the commented-out code.

Reply:
removed